interpretation/src/test/kotlin/com/hartwig/actin/treatment/TreatmentRankingModelTest.kt

```diff
@@ -0,0 +1,251 @@
+                )
+            )
+        )
+        val rank = ranker.rank(patientRecord)
+        assertThat(rank.ranking[0].treatment).isEqualTo("treatment1")
+        assertThat(rank.ranking[0].score).isEqualTo(5100.0)
+    }
+
+    @Test
+    fun `Should sum scores for a single treatment with both benefit and resistance`() {
+        val ranker = TreatmentRankingModel(EvidenceScoringModel())
+        val patientRecord = patientRecord(
+            createVariant(
+                treatmentEvidence = treatmentEvidence(
+                    cancerTypeMatchApplicability = CancerTypeMatchApplicability.SPECIFIC_TYPE,
+                    isCategoryEvent = false,
+                    approvalStage = EvidenceLevelDetails.FDA_APPROVED,
+                    hasBenefit = true,
+                    treatment = "treatment1"
+                )
+            ), createVariant(
+                treatmentEvidence =
+                treatmentEvidence(
+                    cancerTypeMatchApplicability = CancerTypeMatchApplicability.SPECIFIC_TYPE,
+                    isCategoryEvent = false,
+                    approvalStage = EvidenceLevelDetails.GUIDELINE,
+                    hasBenefit = false,
+                    treatment = "treatment1",
+                )
+            )
+        )
+        val rank = ranker.rank(patientRecord)
+        assertThat(rank.ranking[0].treatment).isEqualTo("treatment1")
+        assertThat(rank.ranking[0].score).isEqualTo(100.0)
+    }
+
+    @Test
+    fun `Should diminish returns single treatment, same tumor type, variant`() {
+        val ranker = TreatmentRankingModel(EvidenceScoringModel())
+        val patientRecord = patientRecord(
+            createVariant(
+                treatmentEvidence = treatmentEvidence(
+                    cancerTypeMatchApplicability = CancerTypeMatchApplicability.SPECIFIC_TYPE,
+                    isCategoryEvent = false,
+                    approvalStage = EvidenceLevelDetails.GUIDELINE,
+                    hasBenefit = true,
+                    treatment = "treatment1"
+                )
+            ), createVariant(
+                treatmentEvidence = treatmentEvidence(
+                    cancerTypeMatchApplicability = CancerTypeMatchApplicability.SPECIFIC_TYPE,
+                    isCategoryEvent = false,
+                    approvalStage = EvidenceLevelDetails.FDA_APPROVED,
+                    hasBenefit = true,
+                    treatment = "treatment1",
+                )
+            )
+        )
+        val rank = ranker.rank(patientRecord).ranking
+        assertThat(rank[0].treatment).isEqualTo("treatment1")
+        assertThat(rank[0].score).isEqualTo(2950.0)
+    }
+
+
+    private fun patientRecord(
+        vararg variants: Variant
+    ) = TestPatientFactory.createProperTestPatientRecord().copy(
+        molecularHistory = MolecularHistory(
+            listOf(
+                TestMolecularFactory.createExhaustiveTestMolecularRecord().copy(
```

Comment:
I would use the _minimal_ test records here since you ideally want records with _only_ the drivers you pass in?